fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/InclusionQuerySegmentAggregator.java

```diff
@@ -232,13 +242,14 @@ private void executeIncludeSubQuery(StringBuilder queryString, InclusionParamete
         //The subquery should return a list of strings in the FHIR Reference String value format 
         //(e.g. {@code "Patient/<resource_id>"})
         SqlQueryData subQueryData = new SqlQueryData(subQueryString.toString(), bindVariables);
+
         boolean isFirstItem = true;
         for (String strValue : this.resourceDao.searchStringValues(subQueryData)) {
             if (!isFirstItem) {
                 queryString.append(COMMA);
             }
             if (strValue != null) {
-                queryString.append(QUOTE).append(strValue).append(QUOTE);
+                queryString.append(QUOTE).append(SqlParameterEncoder.encode(strValue)).append(QUOTE);
```

Comment:
what the SqlParameterEncoder.encode is used for here?

Reply:
avoid secondary injection problems when self assembling a query. https://haiderm.com/second-order-sql-injection-explained-with-example/